### PR TITLE
Add basic optional logger class

### PR DESCRIPTION
**Pull request by drewdeponte**
This PR includes the basic optional logger and the following methods.
- `add`
- `log`
- `info`
- `warn`
- `debug`
- `error`
- `fatal`
- `unknown`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ instance as `received_logger`.
 ```ruby
 require 'optional_logger'
 
-logger = OptionalLogger.new(received_logger)
+logger = OptionalLogger::Logger.new(received_logger)
```
Comment (RyanHedges) [inline README.md:63]: alternative name ideas:
`OptLog::Logger.new(received_logger)`
`OptionLog::Logger.new(received_logger)`

File: spec/optional_logger/logger_spec.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    context 'when the optional logger was constructed with a logger' do
+      let(:logger) { double('logger') }
+      subject { described_class.new(logger) }
+
+      it 'returns the logger it was constructed with' do
+        expect(subject.wrapped_logger).to eq(logger)
+      end
+    end
+
+    context 'when the optional laggor was constructed with a nil' do
```
Comment (RyanHedges) [inline spec/optional_logger/logger_spec.rb]: laggor

File: README.md (hunk 1 above)
Comment (cyphactor) [inline README.md]: Nope, can't buy into either of those as they both seem like entities so it sounds like it is a logger of options not an optional logger.

Comment (drewdeponte) [reply README.md:63]: I can't buy into either of those as they both seem like entities so it sounds like it is a logger of options not an optional logger.

I am not a fan of the `Logger::Logger` which is what I am guessing you are trying to get away from. So, I am open to other suggestions.

**Comment by RyanHedges** [comment]
👍

**Pull request merged**